packages/app/utils/token.ts

```diff
@@ -1,8 +1,31 @@
+const getZeroDecimalsCount = (value: number | string) => {
+  if (!value.toString().includes(".")) return 0;
+
+  const firstDecimalZeroIndex = value
+    .toString()
+    .split(".")[1]
+    .split("")
+    .findIndex((digit: string) => digit >= "1");
+
+  return firstDecimalZeroIndex === -1 ? 0 : firstDecimalZeroIndex;
+};
+
 export const addressShortner = (address: string) =>
   address.slice(0, 4) + "..." + address.slice(-4);
 
-export const formatTokenValue = (value: number, custom?: number) => {
+export const formatTokenValue = (
+  value: number | string,
+  decimals: number = 4
+) => {
   if (value === 0) return value.toFixed(2);
 
-  return custom ? value.toFixed(custom) : value.toFixed(4);
+  const nonRelevantZeroes = getZeroDecimalsCount(value);
+  const relevantZeroDecimals = new Array(decimals - 1).fill(0).join("");
+
+  const formattedValue =
+    nonRelevantZeroes >= decimals
+      ? `< 0.${relevantZeroDecimals}1`
+      : Number(value).toFixed(decimals);
+
+  return formattedValue;
 };
```

Comment:
We can dumb down the logic as long as we always use 4 decimals, but the idea for `getZeroDecimalsCount` is to get the number of zeroes before the first relevant magnitude. It's used mainly to format the dust values.

Now that I check it, the function is wrong, because that format should apply only for near zero values and not for any value.

WDYT? For now, it's working with any given amount of decimals (so if we eventually need to format tokens with 6 decimal places we can do it too.

Reply:
The new change I just pushed would yield
`1.000000000000000001` -> `1.0000`
`0.000000000000000001` -> `< 0.0001`